bootstrap/handlers/externalmqtt.go

```diff
@@ -0,0 +1,171 @@
+//
+// Copyright (C) 2022 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package handlers
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/messaging"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/startup"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+)
+
+type ExternalMQTT struct {
+	onConnectHandler mqtt.OnConnectHandler
+}
+
+func NewExternalMQTT(onConnectHandler mqtt.OnConnectHandler) *ExternalMQTT {
+	return &ExternalMQTT{onConnectHandler: onConnectHandler}
+}
+
+func (e *ExternalMQTT) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, startupTimer startup.Timer, dic *di.Container) bool {
+	lc := container.LoggingClientFrom(dic.Get)
+	configuration := container.ConfigurationFrom(dic.Get)
+
+	brokerConfig := configuration.GetBootstrap().ExternalMQTT
+	topics := brokerConfig.SubscribeTopics
+	// TODO: remove first check and update error log in EdgeX 3.0
+	if len(strings.TrimSpace(topics)) == 0 && len(brokerConfig.Topics) == 0 {
+		lc.Errorf("missing SubscribeTopics for external MQTT connection. Must be present in [MessageQueue.External] section")
+		return false
+	}
+
+	_, err := url.Parse(brokerConfig.Url)
+	if err != nil {
+		lc.Errorf("invalid MQTT Broker Url '%s': %s", brokerConfig.Url, err.Error())
+		return false
+	}
+
+	opts := mqtt.NewClientOptions()
+	opts.AddBroker(brokerConfig.Url)
+	opts.SetClientID(brokerConfig.ClientId)
+	opts.SetOnConnectHandler(e.onConnectHandler)
+	opts.SetAutoReconnect(brokerConfig.AutoReconnect)
+	opts.KeepAlive = brokerConfig.KeepAlive
+	if len(brokerConfig.ConnectTimeout) > 0 {
+		duration, err := time.ParseDuration(brokerConfig.ConnectTimeout)
+		if err != nil {
+			lc.Errorf("invalid MQTT ConnectTimeout '%s': %s", brokerConfig.ConnectTimeout, err.Error())
+			return false
+		}
+		opts.SetConnectTimeout(duration)
+	}
+
```

Comment:
Missing setting of `RetryDuration` and `RetryInterval` like is done here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/internal/trigger/mqtt/mqtt.go#L112-L117

Reply:
Do we still need them if we're using the startup duration/interval ?

Reply:
Ahh, right. Nope not needed in this handler.